Media/Manager/MediaStorageManager.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace OpenOrchestra\Media\Manager;
+
+use OpenOrchestra\Media\Exception\BadFileException;
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * Class MediaStorageManager
+ */
+class MediaStorageManager implements MediaStorageManagerInterface
+{
+    protected $fileSystem;
+    protected $mediaDomain;
+    protected $mediaDirectory;
+
+    /**
+     * @param Filesystem            $fileSystem
+     * @param string                $mediaDomain
+     * @param string                $mediaDirectory
+     */
+    public function __construct(
+        Filesystem $fileSystem,
+        $mediaDomain,
+        $mediaDirectory
+    ){
+        $this->fileSystem = $fileSystem;
+        $this->mediaDomain = $mediaDomain;
+        $this->mediaDirectory = $mediaDirectory;
+    }
+
+    /**
+     * Upload $filePath file with the key $key
+     * 
+     * @param string  $key
+     * @param string  $filePath
+     * @param boolean $deleteAfterUpload
+     *
+     * @throws BadFileException
+     */
+    public function uploadFile($key, $filePath, $deleteAfterUpload = true)
+    {
+        if (is_dir($filePath) && false === is_dir($this->mediaDirectory)) {
+            throw new BadFileException();
+        }
+        $targetFile = $this->computePath($key);
+        $this->fileSystem->copy($filePath, $targetFile);
+
+        if ($deleteAfterUpload) {
```

Comment:
@alavieille : is these feature still usefull ? It was designed in case of upload to a distant storage like Amazon S3 to keep a copy on the local filesystem, but as the upload only put the file on the local filesystem now, I think it's useless for now.